src/pages/journal/journal.effects.ts

```diff
@@ -45,6 +45,8 @@ export class JournalEffects {
         .pipe(
           map((journalData: ExaminerWorkSchedule) => this.slotProvider.detectSlotChanges(slots, journalData)),
           map((slots: any[]) => groupBy(slots, this.slotProvider.getSlotDate)),
+          map((slots: {[k: string]: SlotItem[]}) => this.slotProvider.extendWithEmptyDays(slots)),
+          map((slots: {[k: string]: SlotItem[]}) => this.slotProvider.getRelevantSlots(slots)),
```

Comment:
Can you foresee using these two methods of the slot provider independently, or are they completely coupled? If they are coupled, perhaps you could refactor to only have to do a single mapping here?